dhall-docs/README.md

```diff
@@ -97,6 +60,281 @@ dhall-docs --input . --output-link ${OTHER_LINK}
 
 For more information about the tool, check the `--help` flag.
 
+
+## Documenting your packages
+
+The markup language of the documentation is [CommonMark], which is a strict
+Markdown flavour. `dhall-docs` uses [`mmark`], a package that parses
+[CommonMark] and transforms it to HTML. This package follows (almost)
+this specification.
+
+Documenting your package is essentially writing comments on your source code,
+although there are some format rules that these needs to obey in order to work
+properly with `dhall-docs`. In a single file you can use any kind of comments.
+
+You can see examples of writing documentation on the [tests](./tasty/data/package)
+folder.
+
+### Block comments
+
+Normal block comments in Dhall starts with `{-` and ends with `-}`. For example:
+
+```dhall
+{- foo
+     bar
+       baz
+-}
+```
+
+A `dhall-docs` block comment needs to start with `{-|<NEW_LINE>`.
+That is, the actual documentation starts on the following line. The closing brace
+doesn't need any special treatment. This let you separate your own internal block
+comments from the ones that you want `dhall-docs` to output in the final markup.
+
+[CommonMark] supports some features that are sensitive to indentation (e.g.
+[indented-code-blocks]). On a `dhall-docs` block comment, the indentation is
+determined by the column of the `{` character from the opening brace. For
+example, on the following block comment:
+
+```dhall
+{-|
+foo
+  bar
+    baz
+-}
+```
+
+... and the following:
+
+```dhall
+              {-|
+              foo
+                bar
+                  baz
+                        -}
+```
+
+
+... will extract the contents of the comment, stripping the `{` leading whitespace
+from all other lines, returning the following text:
+
+```
+foo
+  bar
+    baz
+```
+
+The indentation level of a line is the number of whitespace characters (i.e. `' '`) from
+the first column until the column of the `{`. The rest of the whitespace that
+is not stripped will be preserved for further manipulations (i.e. converting
+[CommonMark] to HTML).
+
+A block comment that has a line with less indentation than the `{` character
+column one is not a valid `dhall-docs` comment. The following:
+
+```dhall
+    {-|
+    foo
+  bar
+        baz -}
+```
+
+although it's a valid Dhall block comment, it's not valid for `dhall-docs`. The
+same applies for block comments that starts on the same line as the opening
+brace. The following:
+
+```dhall
+{-| foo
+    bar
+    -}
+```
+
+is invalid.
+
+See that the closing brace (`-}`) can be in a new line or in the last one:
+`dhall-docs` will, in any case, insert a newline character (`\n`) in the last
+line. The following example:
+
+```dhall
+{-|
+foo-}
+```
+
+and
+
+```dhall
+{-|
+foo
+-}
+```
+
+will both extract `"foo\n"`.
```

Comment:
Also, is there a difference between

```
{-|
foo
-}
```

and

```
{-|
foo

-}
```

?

Reply:
I'd say they are different. Or at least the tool should not strip the newlines. My reasoning is that Markdown is a little sensitive to whitespaces, and after thinking a little it would be better to do the less whitespace stripping on the text. The only exception would be the prefix whitespace for indentation, that is mostly discussed on the proposal.